### ISSUE TITLE
Last epoch issue on CosineAnnealingDecay

### ISSUE THREAD

**Issue by mingxu1067** [PaddlePaddle/Paddle#43004]
### bug描述 Describe the Bug

The lr got with `last_epoch=1` set in `CosineAnnealingDecay` is not consisent with `last_epoch=-1` and twice `step()` called.
Here is a reproce python script:
```python
import paddle

t_max = (90-5)*626
lrc = paddle.optimizer.lr.CosineAnnealingDecay(
            learning_rate=2.048,
            T_max=t_max,
            eta_min=0.0,
            last_epoch=-1)

print("Initial LR: ", lrc.get_lr())
lrc.step()
print("Step once LR: ", lrc.get_lr())
lrc.step()
print("Step twice LR: ", lrc.get_lr())
lrc.step()
print("Step three times LR: ", lrc.get_lr())

lrc2 = paddle.optimizer.lr.CosineAnnealingDecay(
            learning_rate=2.048,
            T_max=t_max,
            eta_min=0.0,
            last_epoch=1)
print("LR with last epoch=1 :", lrc2.get_lr())
```
and results:
```
Initial LR:  2.048
Step once LR:  2.0479999964304527
Step twice LR:  2.0479999875065835
Step three times LR:  2.047999975013167

LR with last epoch=1 : 2.047999989291357
```

From results, the lr with `last_epoch=1` is ` 2.047999989291357`, which not equals to `Step twice LR:  2.0479999875065835`, even not one of the lr from `CosineAnnealingDecay` with `last_epoch=-1`

### 其他补充信息 Additional Supplementary Information

PaddlePaddle: v2.2.2

**Comment by esythan** [comment]
我们已经收到了你的问题，正在和相关RD同学沟通解决，有结论会第一时间回复你。

**Comment by zhwesky2010** [comment]
@mingxu1067 您好，这个api由于用的递推公式实现的，直接使用last_epoch来恢复训练有些问题，比如指定last_epoch=2时，此时上一个lr还是初始值。所以计算有问题，可以使用scheduler.step(epoch=2)、scheduler.step(epoch=3)这样来实现

**Comment by mingxu1067** [comment]
Workaround verified.

```
Initial LR:  2.048
Step once LR:  2.0479999964304527
Step twice LR:  2.0479999875065835
Step three times LR:  2.047999975013167

LR with last epoch=1 : 2.047999989291357

LR with step(epoch=2) : 2.0479999875065835
```

Would Paddle have a plan to fix this? Or that is an acceptable miss-matching?

**Comment by zhwesky2010** [comment]
> Workaround verified.
> 
> ```
> Initial LR:  2.048
> Step once LR:  2.0479999964304527
> Step twice LR:  2.0479999875065835
> Step three times LR:  2.047999975013167
> 
> LR with last epoch=1 : 2.047999989291357
> 
> LR with step(epoch=2) : 2.0479999875065835
> ```
> 
> Would Paddle have a plan to fix this? Or that is an acceptable miss-matching?

At present, there is no plan to repair it. This problem occurs for all recursive formulas. May throw an exception when set  `last_epoch` for this scheduler in future.
